### PR TITLE
Test EIPs DSL methods #2628

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

File: integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipResource.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public String route(String body, @PathParam("route") String route) {
+        return producerTemplate.requestBody("direct:" + route, body, String.class);
+    }
+
+    @Path("/result/{name}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @GET
+    public String mock(@PathParam("name") String name, @PathParam("count") int count, @PathParam("timeout") int timeout) {
```
Comment (aldettinger) [inline integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipResource.java:56]: Looks count and timeout are not used ?

File: integration-test-groups/foundation/eip/pom.xml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-log-deployment</artifactId>
```
Comment (aldettinger) [inline integration-test-groups/foundation/eip/pom.xml:89]: Not sure log is necessary ?

**Comment by aldettinger** [comment]
ok, the current state is almost good to go :+1:

**Comment by ppalaga** [comment]
Thanks for the suggestions, @aldettinger, I'll adjust the code. I drafted it this only to consult something with Claus. More tests are coming.

**Comment by ppalaga** [comment]
Not complete yet. Let's better review and merge before gets too big and outdated.

File: integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipRoutes.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    @Override
+    public void configure() {
+        from("direct:dynamicRouter")
+                .dynamicRouter(method("myDynamicRouter", "route"));
+
+    }
+
+    @Produces
+    @Unremovable
```
Comment (jamesnetherton) [inline integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipRoutes.java:40]: Nit - If a bean is named, then it's unremovable. So the annotation is not strictly required.

File: integration-tests/file/pom.xml [hunk 4]
```diff
@@ -42,6 +42,10 @@
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-quartz</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-direct</artifactId>
```
Comment (jamesnetherton) [inline integration-tests/file/pom.xml:47]: Best to run the formatting steps so that all of the pseudo dependencies are added.

**Review by jamesnetherton — APPROVED** [review]
Looks ok to me. I left 2 nitpick comments.

File: integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipRoutes.java (hunk 3 above)
Comment (ppalaga) [reply integration-test-groups/foundation/bean/src/main/java/org/apache/camel/quarkus/component/bean/eip/EipRoutes.java:40]: +1 thanks!

File: integration-tests/file/pom.xml (hunk 4 above)
Comment (ppalaga) [reply integration-tests/file/pom.xml:47]: +1, thanks!

**Comment by ppalaga** [comment]
3365696:
* Rebased
* Addressed the suggestions of @jamesnetherton

**Pull request merged**